rfcs/0000-import-expression-ast-node.md

```diff
@@ -0,0 +1,204 @@
+- Repo: `babel/babel`
+- Start Date: 2020-05-01
+- RFC PR: <!-- leave this empty, to be filled in later -->
+- Related Issues: https://github.com/babel/babel/pull/10962
+- Authors: @JLHwung
+- Champion: @JLHwung
+- Implementors: @JLHwung
+
+# Summary
+
+This RFC proposes to change the AST shape of dynamic import `import(moduleName)` in ES2020. A new `ImportExpression` AST node is introduced.
+
+# Motivation
+
+Currently dynamic import is parsed as a `CallExpression`, whose `callee` is a pseudo expression node `Import`. For example, here is the parsed AST of `import("foo")`.
+
+```json
+{
+  "type": "CallExpression",
+  "callee": {
+    "type": "Import"
+  },
+  "arguments": [
+    {
+      "type": "StringLiteral",
+      "value": "foo"
+    }
+  ]
+}
+```
+
+It imposes an incorrect mental model on developers, who may then think of `import()` as a function call. What's more, extending such an AST shape to support stage-1 [module attributes](https://github.com/tc39/proposal-module-attributes) (`import(moduleName, attributes)`) will be awkward if we model `argumets[1]` as the `attributes`. The `arguments[0]` and `arguments[1]` shares the same semantics in AST but it is not the case in module attributes.
```

Comment:
```suggestion
It encourages an incorrect mental model, as developers who may think of `import()` as a function call. It is also problematic when we extend the current AST shape to support stage-1 [module attributes](https://github.com/tc39/proposal-module-attributes) (`import(moduleName, attributes)`) since the arguments (`moduleName` and `attributes` respectively) share the same semantics in the AST.
```